### PR TITLE
Remove rust build phase

### PR DESCRIPTION
**Pull request by dh-ecc**
Addressing #328 

### Overview:

This reorganises `ZCashLightClientKit` to depend on https://github.com/dh-ecc/libzcashlc. `libzcashlc` is the built product of the code previously in the `rust` folder, packaged and able to be consumed (by this repo) by swift package manager and cocoapods.

There are a lot of changes here but most of them are straight forward and can be categorised as below:
1. Remove all rust sources, build related files and built products (they are now part of https://github.com/dh-ecc/libzcashlc).
2. Move `ZCashLightClientKit` -> `Sources/ZCashLightClientKit` (fitting in with swift package manager convention)
3. Update the `ZCashLightClientKit` target in `Package.swift` to add https://github.com/dh-ecc/libzcashlc as a dependency.
4. Update `ZCashLightClientKit.podspect` to use https://github.com/dh-ecc/libzcashlc.
5. Update testing setup:
  a. Move `ZcashLightClientKitTests` -> `Tests/[Target]`, creating three `testTarget`s (and one `target` for shared code), split by external dependencies. specifically whether they require a lightwalletd server running, an internet connection, or _no_ external dependencies.
  b. Update `Package.swift` with these new targets.
  c. Remove testpec declarations in podspec, switch to managing them with SPM.

The updated README should also be used as a reference.

### To be updated separately:

- [x] https://github.com/dh-ecc/libzcashlc should be moved to zcash org, and Podspec / package.swift files should be updated. 
  - [x] (now PR's as https://github.com/zcash-hackworks/zcash-light-client-ffi/pull/1)
- [x] ~We should consider if we want to publish `libzcashlc` on the main cocoapods repo, it would make cocoapods installation easier, it is currently published on https://github.com/dh-ecc/CocoaPodsSpecs.git and so requires an extra `source` declaration in a `Podfile` to install.~ It's fine not to publish, but we should move dh-ecc/CocoaPodsSpecs to zcash org
  - [x] https://github.com/zcash-hackworks/CocoaPodsSpecs/pull/1
- [ ] Update CI to skip any rust environment setup.
- [ ] Tag as `0.13.0`, some references to this in the documentation, assuming it will be the next minor release.

### Testing it's working:

- Open `Package.swift` you should be able to build the framework and run tests.
- Open the `Example/ZcashLightClientSample` project and Build/Run, this tests that SPM integration is working.

### Known issues:

- The CI will fail as we need to update them to remove any rust setup steps.
- The tests are not all passing! Though as far as I can tell, there aren't any failures introduced here that are not currently present in `zcash:master`
 
---

This code review checklist is intended to serve as a starting point for the author and reviewer, although it may not be appropriate for all types of changes (e.g. fixing a spelling typo in documentation).  For more in-depth discussion of how we think about code review, please see [Code Review Guidelines](../blob/main/CODE_REVIEW_GUIDELINES.md).

# Author

- [x] Self-review: Did you review your own code in GitHub's web interface? Code often looks different when reviewing the diff in a browser, making it easier to spot potential bugs.
- [x] Automated tests: Did you add appropriate automated tests for any code changes?
- [x] Code coverage: Did you check the code coverage report for the automated tests?  While we are not looking for perfect coverage, the tool can point out potential cases that have been missed.
- [x] Documentation: Did you update Docs as appropiate? (E.g [README.md](../blob/main/README.md), etc.)
- [x] Run the app: Did you run the app and try the changes? 
- [ ] N/A ~Did you provide Screenshots of what the App looks like before and after your changes as part of the description of this PR? (only applicable to UI Changes)~
- [x] Rebase and squash: Did you pull in the latest changes from the main branch and squash your commits before assigning a reviewer? Having your code up to date and squashed will make it easier for others to review. Use best judgement when squashing commits, as some changes (such as refactoring) might be easier to review as a separate commit.

# Reviewer

- [x] Checklist review: Did you go through the code with the [Code Review Guidelines](../blob/main/CODE_REVIEW_GUIDELINES.md) checklist?
- [x] Ad hoc review: Did you perform an ad hoc review?  _In addition to a first pass using the code review guidelines, do a second pass using your best judgement and experience which may identify additional questions or comments. Research shows that code review is most effective when done in multiple passes, where reviewers look for different things through each pass._
- [x] Automated tests: Did you review the automated tests?
- [x] Manual tests: Did you review the manual tests?_You will find manual testing guidelines under our [manual testing section](../blob/mater/docs/testing/manual_testing)_
- [ ] How is Code Coverage affected by this PR? _We encourage you to compare coverage befor and after your changes and when possible, leave it in a better place. [Learn More...](../blob/master/docs/testing/local_coverage.md)_
- [x] Documentation: Did you review Docs, [README.md](../blob/master/README.md), [LICENSE.md](../blob/master/LICENSE.md), and [Architecture.md](../blob/master/docs/Architecture.md) as appropriate?
- [x] Run the app: Did you run the app and try the changes? While the CI server runs the app to look for build failures or crashes, humans running the app are more likely to notice unexpected log messages, UI inconsistencies, or bad output data.

### CODE REVIEW

**Comment by pacu** [comment]
There are many files that are listed as "renamed" but have no apparent change in their path 
image

File: Package.swift [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+    ],
+    products: [
+        .library(
+            name: "ZcashLightClientKit",
+            targets: ["ZcashLightClientKit"]
+        ),
+    ],
+    dependencies: [
+        .package(url: "https://github.com/grpc/grpc-swift.git", from: "1.0.0"),
+        .package(url: "https://github.com/stephencelis/SQLite.swift.git", from: "0.13.0"),
```
Comment (pacu) [inline Package.swift:18]: this library was 0.12.2 were test ran to verify that the changes did not break anything ?

File: Sources/ZcashLightClientKit/Constants/WalletBirthday+mainnet.swift [hunk 2]
```diff
@@ -479,117 +479,19 @@ extension WalletBirthday {
                 time: 1627091654,
                 tree: "016cc0651c73617064bf6de5962c99ec81fdce76d58ac2aab8b881e3b35991486d01593438b57eb9cfa76d5ea965b77a0304ff46163846c7c1ffb2695419a8dbfd4c1301e9766721a57a35e08763a27244291cb7db2ca67a711fd0569fd78aa675a21e1b019a9908d737043f786d7cdddbb22cb4d026c6c3221d34bd82bf4dc55acdd847640000000001de097ad39006362760b514f69a3aa88c66ada02085de26de23b522167df616320000000001a090ee174239a34a5d684425d09006d238c6075a61c5842d0fc26043f09ccd7001a2b7ee187c7b8ce18ebda8600bed7695b12f7d35ac971ed6ee67184a7ceebd490001b35fe4a943a47404f68db220c77b0573e13c3378a65c6f2396f93be7609d8f2a000125911f4524469c00ccb1ba69e64f0ee7380c8d17bbfc76ecd238421b86eb6e09000118f64df255c9c43db708255e7bf6bffd481e5c2f38fe9ed8f3d189f7f9cf2644"
             )
-        case 1336316 ..< 1340000:
+        case 1336316 ..< BlockHeight.max:
```
Comment (pacu) [inline Sources/ZcashLightClientKit/Constants/WalletBirthday+mainnet.swift:482]: This file is reverting changes

File: Sources/ZcashLightClientKit/Constants/WalletBirthday+testnet.swift [hunk 3]
```diff
@@ -17,320 +17,253 @@ extension WalletBirthday {
...
             )
+            
         default:
             return WalletBirthday(
-                height: 1653741,
-                hash: "0037b39b897bd30532ea0135a2293362fee030c4b071ee4643b9ea980e613ec0",
-                time: 1637275363,
-                tree: "01fa00140317591055501b80be3e94c2b744496e6d67dc001861707b25ddccb00a00100001501c2f90929a108ec2f84d7311524e8e1a10777c91c3bbaa16508ae66cb83173013ef7e369b780797f32cb1740ac6a9c44af7d312e38ece18e5bd6692d3ff484210001a4695cd51c08ab7fc24c2996cbeea4b57233426fadab61463baf7a2785033359000178d02a059f1dd72c16ec63a877287360400ba4e358f76b6bbb1fcddb8bc54e2c01d0b24b33c944586ad94b0ce443f8e6397f66fe8eb530c1658d3118f841e37905000117bce82d9d9f3fd58b4edfb4df6852d3322f451a80bf64351ef9fe7e51e1e21001bab80e68a5c63460d1e5c94ef540940792fa4703fa488b09fdfded97f8ec8a3d00013d2fd009bf8a22d68f720eac19c411c99014ed9c5f85d5942e15d1fc039e28680001f08f39275112dd8905b854170b7f247cf2df18454d4fa94e6e4f9320cca05f24011f8322ef806eb2430dc4a7a41c1b344bea5be946efc7b4349c1c9edb14ff9d39"
+                height: 1507328,
```
Comment (pacu) [inline Sources/ZcashLightClientKit/Constants/WalletBirthday+testnet.swift:263]: this file is reverting changes

File: changelog.md [hunk 4]
```diff
@@ -1,213 +0,0 @@
-# 0.12.0-alpha.11
```
Comment (pacu) [inline changelog.md:1]: why is this file being deleted?

File: docs/Architecture.md [hunk 5]
```diff
@@ -1,6 +0,0 @@
-# Architecture
```
Comment (pacu) [inline docs/Architecture.md:1]: Why is this file being deleted?

File: docs/ci.md [hunk 6]
```diff
@@ -1,56 +0,0 @@
-# Continuous Integration
```
Comment (pacu) [inline docs/ci.md:1]: Why is this file being deleted?

File: responsible_disclosure.md [hunk 7]
```diff
@@ -1,64 +0,0 @@
-This page is copyright The Electric Coin Company, 2021. It is posted in order to conform to this standard: https://github.com/RD-Crypto-Spec/Responsible-Disclosure/tree/d47a5a3dafa5942c8849a93441745fdd186731e6
```
Comment (pacu) [inline responsible_disclosure.md]: Why is this file being deleted?

File: docs/testing/manual_testing/README.md [hunk 8]
```diff
@@ -1,38 +0,0 @@
-# Manual testing
-
```
Comment (pacu) [inline docs/testing/manual_testing/README.md]: Why is this file being deleted?

File: docs/testing/local_coverage.md [hunk 9]
```diff
@@ -1,14 +0,0 @@
-# Gathering Code Coverage
```
Comment (pacu) [inline docs/testing/local_coverage.md]: Why is this file being deleted?

**Review by pacu — CHANGES_REQUESTED** [review]
There are many files being deleted. the fork appears to be out of date.

File: DerivationToolTests/DerivationToolMainnetTests.swift [hunk 10]
```diff
@@ -1,103 +0,0 @@
...
-//  DerivationToolTests.swift
-//  ZcashLightClientKit-Unit-Tests
-//
-//  Created by Francisco Gindre on 10/9/20.
-//
-//swiftlint:disable force_unwrapping
-import XCTest
-import ZcashLightClientKit
-
-class DerivationToolMainnetTests: XCTestCase {
```
[the remark this answers is not in the document]
Comment (Daniel1of1) [reply DerivationToolTests/DerivationToolMainnetTests.swift]: They're back now, this was temporary 😊.

File: Example/ZcashLightClientSample/Podfile [hunk 11]
```diff
@@ -1,15 +1,18 @@
...
   use_frameworks!
   pod "KRProgressHUD"
   # Pods for ZcashLightClientSample
-  pod 'ZcashLightClientKit', :path => '../../', :testspecs => ['Tests', 'DerivationToolTests']
+  pod 'ZcashLightClientKit', :path => '../../'
   pod 'gRPC-Swift', '= 1.0.0'
   pod 'PaginatedTableView'
   pod 'NotificationBubbles'
-  pod 'MnemonicSwift', :git => 'https://github.com/zcash-hackworks/MnemonicSwift.git', :tag => '2.1.0'
+  pod 'MnemonicSwift', '~> 2.0.0'
```
[the remark this answers is not in the document]
Comment (Daniel1of1) [reply Example/ZcashLightClientSample/Podfile]: I've moved from cocoapods to SPM for the Example project dependencies, though I might need to make sure it points to the hackworks fork. From what I can see it's building and running fine currently. Let me know if it should be changed.

File: README.md [hunk 12]
```diff
@@ -1,227 +0,0 @@
-# Zcash iOS Framework
-
-[![Build Status](https://travis-ci.org/zcash/ZcashLightClientKit.svg?branch=master)](https://travis-ci.org/zcash/ZcashLightClientKit)
```
[the remark this answers is not in the document]
Comment (Daniel1of1) [reply README.md]: Back again 😊, and updated accordingly.

File: LICENSE [hunk 13]
```diff
@@ -1,21 +0,0 @@
-MIT License
```
[the remark this answers is not in the document]
Comment (Daniel1of1) [reply LICENSE]: Back again 😊

File: Package.swift (hunk 1 above)
Comment (Daniel1of1) [reply Package.swift:18]: Yep this is working fine, IIRC it was necessary for arm64 simulator support, can double check.

File: Sources/ZcashLightClientKit/Constants/WalletBirthday+mainnet.swift (hunk 2 above)
Comment (Daniel1of1) [reply Sources/ZcashLightClientKit/Constants/WalletBirthday+mainnet.swift]: Thanks! removed the change.

File: Sources/ZcashLightClientKit/Constants/WalletBirthday+testnet.swift (hunk 3 above)
Comment (Daniel1of1) [reply Sources/ZcashLightClientKit/Constants/WalletBirthday+testnet.swift]: Good spot@, removed this change too!

File: changelog.md (hunk 4 above)
Comment (Daniel1of1) [reply changelog.md]: File restored 😊

File: docs/Architecture.md (hunk 5 above)
Comment (Daniel1of1) [reply docs/Architecture.md]: File restored 😊

File: docs/ci.md (hunk 6 above)
Comment (Daniel1of1) [reply docs/ci.md]: File restored 😊

File: responsible_disclosure.md (hunk 7 above)
Comment (Daniel1of1) [reply responsible_disclosure.md]: File restored 😊

File: docs/testing/manual_testing/README.md (hunk 8 above)
Comment (Daniel1of1) [reply docs/testing/manual_testing/README.md]: File restored 😊

File: docs/testing/local_coverage.md (hunk 9 above)
Comment (Daniel1of1) [reply docs/testing/local_coverage.md]: File restored 😊

File: DerivationToolTests/DerivationToolMainnetTests.swift (hunk 10 above)
[the remark this answers is not in the document]
Comment (dh-ecc) [reply DerivationToolTests/DerivationToolMainnetTests.swift]: Tests restored 😄 . See updated PR description for details.

File: README.md (hunk 12 above)
[the remark this answers is not in the document]
Comment (dh-ecc) [reply README.md]: File Restored 👍

File: SWIFTLINT.md [hunk 14]
```diff
@@ -1,126 +0,0 @@
-# Swift Style Guide
```
[the remark this answers is not in the document]
Comment (dh-ecc) [reply SWIFTLINT.md]: File Restored 👍

File: LICENSE (hunk 13 above)
[the remark this answers is not in the document]
Comment (dh-ecc) [reply LICENSE]: File Restored 👍

File: Package.swift (hunk 1 above)
Comment (dh-ecc) [reply Package.swift:18]: Yep, working fine as far as I can tell. IIRC I think this update was necessary for arm64 simulator support, I will double check this.

File: Sources/ZcashLightClientKit/Constants/WalletBirthday+mainnet.swift (hunk 2 above)
Comment (dh-ecc) [reply Sources/ZcashLightClientKit/Constants/WalletBirthday+mainnet.swift]: Thanks for catching this! I've undone the change.

File: Sources/ZcashLightClientKit/Constants/WalletBirthday+testnet.swift (hunk 3 above)
Comment (dh-ecc) [reply Sources/ZcashLightClientKit/Constants/WalletBirthday+testnet.swift]: Thanks for catching this! I've undone the change.

File: changelog.md (hunk 4 above)
Comment (dh-ecc) [reply changelog.md]: File restored! Though probably needs an update for the contents of this PR (though might make sense to add it separately after moving some of the repos we depend into zcash org and then tagging that release)

File: docs/Architecture.md (hunk 5 above)
Comment (dh-ecc) [reply docs/Architecture.md]: File Restored 👍

File: responsible_disclosure.md (hunk 7 above)
Comment (dh-ecc) [reply responsible_disclosure.md]: File Restored 👍

File: docs/ci.md (hunk 6 above)
Comment (dh-ecc) [reply docs/ci.md]: File Restored 👍

File: docs/testing/manual_testing/README.md (hunk 8 above)
Comment (dh-ecc) [reply docs/testing/manual_testing/README.md]: File Restored 👍

File: docs/testing/local_coverage.md (hunk 9 above)
Comment (dh-ecc) [reply docs/testing/local_coverage.md]: File Restored 👍

**Comment by pacu** [comment]
AdvancedReOrgTests passed
image

**Review by pacu — APPROVED** [review]

**Comment by pacu** [comment]
Approved overall. Requesting changes for 
> https://github.com/dh-ecc/libzcashlc should be moved to zcash org, and Podspec / package.swift files should be updated.
but using https://github.com/zcash-hackworks/zcash-light-client-ffi instead.


> It's fine not to publish, but we should move dh-ecc/CocoaPodsSpecs to zcash org

use this repository https://github.com/zcash-hackworks/CocoaPodsSpecs

File: Package.swift [hunk 15]
```diff
@@ -0,0 +1,61 @@
+// swift-tools-version:5.5
+import PackageDescription
+
+let package = Package(
+    name: "ZcashLightClientKit",
+    platforms: [
+        .iOS(.v13),
```
Comment (pacu) [inline Package.swift:7]: Current minimum deployment type for this SDK is 
` s.ios.deployment_target = '12.0'`

**Review by pacu — CHANGES_REQUESTED** [review]
great work, please check minimum deployment target for package file

File: Package.swift (hunk 15 above)
Comment (dh-ecc) [reply Package.swift]: 👍 done in 3b6490c

**Review by pacu — APPROVED** [review]

**Pull request merged**